src/main/java/org/jenkinsci/remoting/nio/FifoBuffer.java

```diff
@@ -339,11 +369,14 @@ public void write(byte[] buf, int start, int len) throws InterruptedException, I
         while (len>0) {
             int chunk;
 
-            synchronized (lock) {
+            final boolean shouldWaitToCleanTheBuffer;
+            synchronized (lock) { 
                 while ((chunk = Math.min(len,writable()))==0) {
-                    if (closed)
-                        throw new IOException("closed during write operation");
-
+                    if (closeRequested) {
```

Comment:
I'm still not comfortable about not having timeout if the buffer is full && does not get freed by other threads